cocos/2d/assets/sprite-frame.ts

```diff
@@ -1316,6 +1410,140 @@ export class SpriteFrame extends Asset {
+            const texh = tex.height;
+            const rect = this.rect;
+            const width = rect.width;
+            const height = rect.height;
+            const rectX = rect.x;
+            const rectY = texh - rect.y - height;
+            const halfWidth = width / 2;
+            const halfHeight = height / 2;
+
+            const l = texw === 0 ? 0 : rectX / texw;
+            const r = texw === 0 ? 1 : (rectX + width) / texw;
+            const t = texh === 0 ? 1 : (rectY + height) / texh;
+            const b = texh === 0 ? 0 : rect.y / texh;
+
+            // left bottom
+            temp_vec3.set(-halfWidth, -halfHeight, 0);
+            this.vertices.rawPosition.push(temp_vec3.clone());
+            this.vertices.uv.push(rectX);
+            this.vertices.uv.push(rectY + height);
+            this.vertices.nuv.push(l);
+            this.vertices.nuv.push(b);
+            this.vertices.minPos.set(temp_vec3);
+            // right bottom
+            temp_vec3.set(halfWidth, -halfHeight, 0);
+            this.vertices.rawPosition.push(temp_vec3.clone());
+            this.vertices.uv.push(rectX + width);
+            this.vertices.uv.push(rectY + height);
+            this.vertices.nuv.push(r);
+            this.vertices.nuv.push(b);
+            // left top
+            temp_vec3.set(-halfWidth, halfHeight, 0);
+            this.vertices.rawPosition.push(temp_vec3.clone());
+            this.vertices.uv.push(rectX);
+            this.vertices.uv.push(rectY);
+            this.vertices.nuv.push(l);
+            this.vertices.nuv.push(t);
+            // right top
+            temp_vec3.set(halfWidth, halfHeight, 0);
+            this.vertices.rawPosition.push(temp_vec3.clone());
+            this.vertices.uv.push(rectX + width);
+            this.vertices.uv.push(rectY);
+            this.vertices.nuv.push(r);
+            this.vertices.nuv.push(t);
+            this.vertices.maxPos.set(temp_vec3);
+
+            this.vertices.triangles.push(0);
+            this.vertices.triangles.push(1);
+            this.vertices.triangles.push(2);
+            this.vertices.triangles.push(2);
+            this.vertices.triangles.push(1);
+            this.vertices.triangles.push(3);
+        }
+        this._updateMeshVertices();
+    }
+
+    // 综合顶点信息、单位信息、锚点、extrude 甚至 customOutline 来生成实际使用的 vertices
+    protected _updateMeshVertices () {
+        //开始生成生成 mesh 要的 Geometry 信息
+        temp_matrix.identity();
+
+        const units = 1 / this._pixelsToUnit;
+        const temp_vec3 = new Vec3(units, units, 1);
+        temp_matrix.scale(temp_vec3);
+        const PosX = -(this._pivot.x - 0.5) * this.rect.width * units;
+        const PosY = -(this._pivot.y - 0.5) * this.rect.height * units;
+        temp_vec3.set(PosX, PosY, 0);
+        temp_matrix.translate(temp_vec3);
+        const vertices = this.vertices!;
+
+        for (let i = 0; i < vertices.rawPosition.length; i++) {
+            const pos = vertices.rawPosition[i];
```

Comment:
Can we just share `rawPositions` and `positions`, convert each position and store back
But we might need to save rawPositions in editor, it's not a big deal, just want to save memory foot print in runtime